File: Niveau 1/7 – Conditions avancées, opérateurs booléens/10 - Zones de couleurs.py
for i in range(int(input())):
    X = int(input())
    Y = int(input())
    if (Y >= 60 and Y <= 70) and ((X >= 15 and X <= 45) or (X >= 60 and X <= 85)):
        print("Dans une zone rouge")
    elif ((0 <= Y <= 10 or 55 <= Y <= 70) and 0 <= X <= 90) or ((0 <= X <= 10 or 85 <= X <= 90) and 0 <= Y <= 70) or (25 <= X <= 50 and 20 <= Y <= 45):
        print("Dans une zone jaune")
    elif ((Y >= 10 and Y <= 55) and (X >= 10 and X <= 85)):
        print("Dans une zone bleue")
    else:
        print("En dehors de la feuille")

File: Niveau 1/7 – Conditions avancées, opérateurs booléens/10 - Zones de couleurs.cpp
#include <iostream>
using namespace std;
int main()
{
  int nbJeton;
  cin >> nbJeton;
  for (int i = 0; i < nbJeton; i++)
  {
    int x, y;
    cin >> x >> y;
    if (x < 0 || x > 90 || y < 0 || y > 70)
    {
      cout << "En dehors de la feuille" << endl;
    }
    else if (x >= 25 && x <= 50 && y >= 20 && y <= 45)
    {
      cout << "Dans une zone jaune" << endl;
    }
    else if (x >= 10 && x <= 85 && y >= 10 && y <= 55)
    {
      cout << "Dans une zone bleue" << endl;
    }
    else if (y >= 60 && ((x >= 15 && x <= 45) || (x >= 60 && x <= 85)))
    {
      cout << "Dans une zone rouge" << endl;
    }
    else
    {
      cout << "Dans une zone jaune" << endl;
    }
  }
}